Refactor e2e tests to use kueue

ChughShilpa opened the pull request:
# Issue link
<https://issues.redhat.com/browse/RHOAIENG-5239>
<!-- If the issue is closed with this PR enter 'Closes #<issue_number>' -->

# What changes have been made
Refactored e2e tests to use kueue and not mcad. 
Added steps to create kueue resources such as cluster queue, resource flavour and local queue.

# Verification steps
Execute e2e tests following https://github.com/project-codeflare/codeflare-sdk/blob/main/docs/e2e.md 


## Checks
- [X] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [X] Manual tests
   - [ ] Testing is not required for this change

<!-- You can find out information on the review process at this link https://github.com/project-codeflare/codeflare/blob/develop/CONTRIBUTING.md#getting-feedback-on-your-contribution -->

sutaakar commented on tests/e2e/kueue_resources_setup.sh:
```diff
@@ -0,0 +1,55 @@
+    spec:
+      namespaceSelector: {}
+      resourceGroups:
+      - coveredResources: ["cpu", "memory", "pods", "nvidia.com/gpu"]
```
IMHO there is no need to cover number of pods.

sutaakar commented:
`tests/e2e/mnist_raycluster_sdk_test.py` seems to be a duplication of one test case in `tests/e2e/mnist_raycluster_sdk_oauth_test.py` . IMHO `mnist_raycluster_sdk_test.py` can be removed.

Fiona-Waters reviewed (changes requested):
Just a couple of nitpicks

ChughShilpa commented:
> `tests/e2e/mnist_raycluster_sdk_test.py` seems to be a duplication of one test case in `tests/e2e/mnist_raycluster_sdk_oauth_test.py` . IMHO `mnist_raycluster_sdk_test.py` can be removed.

Removed `mnist_raycluster_sdk_test.py` 
Updated dedicated test for kind cluster `assert_jobsubmit_withoutlogin_kind`

ChughShilpa commented on tests/e2e/mnist_raycluster_sdk_oauth_test.py:
```diff
@@ -47,19 +47,21 @@ def run_mnist_raycluster_sdk_oauth(self):
 
-        cluster.wait_ready()
+    # Uncomment cluster.wait_ready() once the issue https://issues.redhat.com/browse/RHOAIENG-3981 is resolved
+    #    cluster.wait_ready
```
Added cluster.wait_ready again as the issue has been fixed

Fiona-Waters commented:
Ran test against an openshift cluster and it passes as expected.
/lgtm

sutaakar commented on docs/e2e.md:
```diff
@@ -10,8 +10,13 @@ Pre-requisite for KinD clusters: please add in your local `/etc/hosts` file `127
-  make deploy -e IMG=quay.io/project-codeflare/codeflare-operator:v1.1.0
   make setup-e2e
+
+  # For running tests locally on Kind cluster, we need to disable oauth (which is enabled by default) by setting the boolean value as false
+  [Disable oauth](https://github.com/project-codeflare/codeflare-operator/blob/main/main.go#L117)
```
This can be done by modifying `codeflare-operator-config` ConfigMap, disabling `rayDashboardOAuthEnabled` and restarting CFO.

Fiona-Waters commented:
I have reviewed again with most recent changes on an openshift cluster - test passed as expected.
/lgtm

sutaakar reviewed (approved):
/lgtm
PR check errors are probably caused by RBAC issues, will be investigated in separate PR
